assets/js/modules/analytics/datastore/tags.js

```diff
@@ -114,68 +163,78 @@ export const controls = {
 		return getExistingTag( 'analytics' );
 	},
 	[ FETCH_TAG_PERMISSION ]: ( { payload: { propertyID } } ) => {
-		return API.get( 'modules', 'analytics', 'tag-permission', { propertyID } );
+		return API.get( 'modules', 'analytics', 'tag-permission', { propertyID }, {
+			useCache: false,
+		} );
 	},
+	[ WAIT_FOR_EXISTING_TAG ]: createRegistryControl( ( registry ) => () => {
+		const isExistingTagLoaded = () => registry.select( STORE_NAME ).getExistingTag() !== undefined;
+		if ( isExistingTagLoaded() ) {
+			return true;
+		}
+
+		return new Promise( ( resolve ) => {
+			const unsubscribe = registry.subscribe( () => {
+				if ( isExistingTagLoaded() ) {
+					unsubscribe();
+					resolve();
+				}
+			} );
+		} );
```

Comment:
Yeah I think there is an abstraction to be made for sure. Also, this needs to be a bit more robust to handle an error, or anything that would otherwise cause it to wait forever. I think even potentially setting a timeout for a maximum wait time would be good as well.

Regarding `waitForSelector`, we kind of already have this in `test-utils` with `subscribeUntil`. Maybe that's worth moving into the src since it would be useful for more than just tests? 

@tofumatt - I'd be interested to hear your thoughts on the above as well.